Задание - "Список покупок" (Task 279830)

Sunderik opened the pull request:
Pull Request для задания после первой итерации работы над ним.

Выполнена работа над заданием. Отдельным [коммитом ](https://github.com/Sunderik/surf_course/commit/ef8311d2beb991f48212172620b143923075823f) оформлен ответ на вопрос в комментариях к модели.
Для всего кода добавлены комментарии.

В фидбеке хотелось бы получить не только замечания по реализации проекта, но и оценку правильности формирования его структуры, архитектуры и чистоты кода, или примеров где ООП, SOLID и чистая архитектура явно нарушены.

Sunderik committed:
feat: добавил базовые модели из ТЗ и источник получения данных

Sunderik committed:
feat: настроил тему для приложения и создал нужные расширения

Sunderik committed:
feat: выполнение задания

Sunderik committed:
feat: добавлен ответ на вопрос в комментариях

Sunderik committed:
feat: заполнена форма описания проекта и добалены медиафайлы с примерами работы программы

DmitrDomrachev commented on task-279830/lib/core/domain/logic/prodicts_list_sorting_mixin.dart:
```diff
@@ -0,0 +1,91 @@
+          break;
+        }
+      case SortingTypesEnum.none:
+      default:
+        {
+          break;
+        }
+    }
```
не совсем понятно, зачем добавил default. если мы обрабатываем enum в свиче, то невозвомжно явно не обработать все варианты, так как анализатор кода явно подсветит этот момент.
также сейчас не нужно писать break после каждого свича, глянь доку по этому поводу

Sunderik replied:
Вообще убрал этот метод при переработке кода с enum на классы

DmitrDomrachev commented on task-279830/lib/core/domain/logic/prodicts_list_sorting_mixin.dart:
```diff
@@ -0,0 +1,91 @@
+  }
+
+  /// Отправить сигнал контроллеру об изменении типа сортировки на [sortingType].
+  void _sinkSelectedSortingType(SortingTypesEnum sortingType) {
+    _selectedSortingType = sortingType;
+    _selectedSortingTypeController?.sink.add(sortingType);
+  }
```
https://github.com/stanlag/surf_course/pull/11
вот тут пример реализации, в которой не нужно явно передавать тип сотировки для определения типа отображения (по категориям или нет)

Sunderik replied:
Надеюсь правильно понял ваше замечание, и переделал работу выбора сортировки с enum на классы.

DmitrDomrachev commented on task-279830/lib/core/library/product_item_widget.dart:
```diff
@@ -0,0 +1,173 @@
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 16.0),
```
EdgeInsets.symmetric(vertical: 16)
не нужно десятичную часть явно указывать, если она равна нулю. на это даже есть отдельное правило линтера

Sunderik replied:
Во всех отступах убрал дробные части

DmitrDomrachev reviewed (commented):
Очень крутая реализация, задание принимаю. Выше оставил пару незначительных комментариев по оформлению

Sunderik committed:
fix: исправление замечаний к PR

- Убрал break из switch - case
- Убрал десятичные части значений отступов в pading
- Переработал работу с типами сортировки с enum на классы
